src/gcp/proto.ts

```diff
@@ -60,3 +60,44 @@ export function renameIfPresent<Src, Dest>(
   }
   dest[destField] = converter(src[srcField]);
 }
+
+export function fieldMasks(
+  object: Record<string, any>,
+  includeEmptyValues: boolean = false
+): string[] {
+  const masks: string[] = [];
+  for (const key of Object.keys(object)) {
+    fieldMasksHelper(key, object[key], includeEmptyValues, masks);
+  }
+  return masks;
+}
+
+function fieldMasksHelper(
+  prefix: string,
+  cursor: any,
+  includeEmptyValues: boolean,
```

Comment:
I share your reticence on this - it feels like this would be very confusing behavior to someone who isn't familiar with the codebase. However, I do like much simpler this makes our API methods, and I can't think of a better way to 'hide' the complexity of defining field masks when unsetting fields when updating.

This param name is a bit misleading right now, tho - `includeEmptyValues` has no effect on whether empty arrays or protobuf.Empty values are included. Something like "includeNullOrUndefinedValues" would be more verbose but clearer, imo.